### DISCUSSION TITLE
Why DateTimeOffset.Date returns DateTime instead of DateTimeOffset?

### DISCUSSION THREAD

**Discussion by christallire** [dotnet/runtime#98626] [Q&A]
It needs to be more straightforward. It's not foolproof enough.
I was scratching my head because it changed the variable's timezone unnoticed.

for an example:
```
        var truncatedDate = DateTimeOffset.UtcNow.Date;
        DateTimeOffset codeUsingDateTimeOffsetSomewhere = truncatedDate;
```

**Comment by Clockwork-Muse** [comment]
Because at the time `DateTimeOffset` was created, the `Date` class didn't exist yet.

Note that by removing the time information you're also removing anything that would allow you to tie back to a time zone anyhow, so a change in time zone is to be expected.

**Reply by colejohnson66** [reply]
That’s not the question. Why is a `DateTimeOffset` property returning something that *isn’t* a `DateTimeOffset`?

**Reply by Clockwork-Muse** [reply]
Because an actual **date** domain value has neither time nor timezone information, and that's what the property is intended to provide.  The best they could do at the time, lacking a date-specific domain type like `DateOnly`, was use the existing `DateTime` type, and set a couple of properties so it would be "regionless" (however poorly that works in some situations).

For comparison, the [equivalent signature in NodaTime does use a `DateOnly` equivalent type](https://nodatime.org/3.1.x/api/NodaTime.OffsetDateTime.html#NodaTime_OffsetDateTime_Date):
```csharp
public readonly struct OffsetDateTime {
    ....
    public readonly LocalDate Date { get; }
}
```
